File: C++/995. Minimum Number of K Consecutive Bit Flips.cpp
class Solution {
public:
    int minKBitFlips(vector<int>& nums, int k) {
        deque<int> dq; 
        int ans = 0, flip = 0;         
        for (int i = 0; i < size(nums); ++i) {
            if (size(dq) && dq.front() == i) {
                dq.pop_front(); 
                flip ^= 1; 
            }
            if (nums[i] == flip) {
                if (size(nums)-i < k) return -1; 
                ++ans; 
                flip ^= 1; 
                dq.push_back(i+k); 
            }
        }
        return ans; 
    }
};

File: JAVA/995. Minimum Number of K Consecutive Bit Flips.java
class Solution {
    public int minKBitFlips(int[] nums, int k) {        
        int count = 0;
        int currFlipCount = 0;
        for (int i = 0; i < nums.length; i++) {
            if (i - k >= 0 && nums[i - k] == 2) {
                currFlipCount -= 1;
            }
            if ((nums[i] + currFlipCount) % 2 == 0) {
                if (i + k > nums.length) return -1;
                count += 1;
                currFlipCount += 1;
                nums[i] = 2;
            }
        }
        return count;
    }
}
